tsc/scripts/git_count.py

```diff
@@ -0,0 +1,495 @@
+
+            # Update github id to company mapping db
+            add_entry_to_id2company(name, comp)
+            update_company_commit_counts(comp, ccount)                    
+        debugprint("{} : {} : {} : {}".format(name, ccount, comp, email))
+
+    os.chdir("..")
+    debugprint("CCT: {}".format(cct))
+
+#
+# Util function for sorting json 
+#
+def get_sortest_key(a: dict, o: dict):
+    v = None
+    k = None
+    for key, value in a.items():
+        if v is None:
+            v = value
+            k = key
+            continue
+        if v < value:
+            v = value
+            k = key
+    o.update({k: v})
+    a.pop(k)
+    if a:
+        get_sortest_key(a, o)
+    else:
+        return
+
+#
+# Main entry
+#
+def main():
+
+    #
+    # Global variables
+    #
+    global id2company
+    global companys
+    global debug
+
+    #
+    # parser flags
+    #
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--debug", action="store_true",
+                        help="set debug flag")
+    parser.add_argument("--check_ids", action="store_true",
+                        help="check github id to company")
+    parser.add_argument("--idfile", type=str, help="github id file", default="")    
+    parser.add_argument("--pull_workspace", action="store_true",
+                        help="pull workspace")
+    parser.add_argument('--years', type=int, help="the number of years since now")
+    parser.add_argument("--token", type=str, help="github access token", default="")
+    args = parser.parse_args()
+
+    debug = args.debug
+
+    #
+    # Pull workspace only
+    #
+    if args.pull_workspace:
+         pull_repros()
+         return
+     
+    #
+    # Set up gtoken
+    #
+    if args.token != "":
+        global gtoken
+        gtoken = args.token
+        
+    print("Debug flag {}".format(debug))
+
+    #
+    # Check id to company via profile
+    #
+    if args.check_ids and args.idfile != "":
+        check_ids_to_company(args.idfile)
+        return
+    
+    #
+    # Total is used to store all repro's commit info
+    #
+    total = {}
+
+    #
+    # github to company mapping
+    #
+    id2company={}
+
+    #
+    # Set up pp
+    #
+    pp = pprint.PrettyPrinter(width=41, compact=True)
+
+    #
+    # Read cached github id to company mapping if it is available
+    #
+    idfname = "/tmp/id2comp.json"
+    if os.path.isfile(idfname):
+        with open(idfname, 'r') as openfile:
+            # Reading from json file
+            try:
+                id2company = json.load(openfile)
+                debugprint("Read back id2commpany")
+            except:
+                id2company={}
+
+    num_repro = 0
+    years = 5
+    #
+    # loop through all repros
+    #
+    for repro in repros:
+        num_repro = num_repro +1
+        companys={}
+
+        #
+        # Collect commit info from each repro
+        #
+        count_repro_commits_via_short_log(repro, years)
```

Comment:
Besides code PR, there are many other contribution to be accounted for SII like reviews, issue triage etc, I assume they will be updated in the next patchset